build.c

```diff
@@ -539,7 +560,7 @@ build(void)
 	nstarted = 0;
 	for (;;) {
 		/* start ready edges */
-		while (work && numjobs < buildopts.maxjobs && numfail < buildopts.maxfail) {
+		while (work && (numjobs == 0 || loadavglow) && numjobs < buildopts.maxjobs && numfail < buildopts.maxfail) {
```

Comment:
Rather than this `loadvglow` global, I think I'd prefer if we just had the `queryload` function return the load as a double. Then at the start of this loop, do something like

```
maxjobs = buildopts.maxload && queryload() > buildopts.maxload ? 1 : buildopts.maxjobs;
```

and use `maxjobs` instead of `buildopts.maxjobs` in the while loop. What do you think of that approach?

Reply:
Makes sense, we can even mimic Ninja's load average logic and use something like:

     (numjobs == 0 || buildopts.maxload <= 0.0 || queryload() < buildopts.maxload)

Reply:
I think both logic works out the same, right? We start a job if there are none currently running, or if we are under the max load and max jobs.

The reason I suggested the former is to avoid the really long condition.

Reply:
Changed to add a `maxjobs` variable and an `if` statement. Going to mark this as resolved, but feel free to comment on the newer version if you disagree.